decidim-system/app/forms/decidim/system/update_organization_form.rb

```diff
@@ -4,108 +4,39 @@
-          end
-        end
-      end.flatten(1)
-
-      jsonb_attribute :omniauth_settings, OMNIATH_PROVIDERS_ATTRIBUTES
-
-      validates :name, :host, :users_registration_mode, presence: true
-      validate :validate_organization_uniqueness
-      validates :users_registration_mode, inclusion: { in: Decidim::Organization.users_registration_modes }
-
-      def map_model(model)
-        self.secondary_hosts = model.secondary_hosts.join("\n")
-        self.omniauth_settings = (model.omniauth_settings || {}).transform_values do |v|
-          Decidim::OmniauthProvider.value_defined?(v) ? Decidim::AttributeEncryptor.decrypt(v) : v
-        end
-        self.file_upload_settings = FileUploadSettingsForm.from_model(model.file_upload_settings)
-      end
-
-      def clean_secondary_hosts
-        return unless secondary_hosts
-
-        secondary_hosts.split("\n").map(&:chomp).select(&:present?)
-      end
-
-      def clean_available_authorizations
-        return unless available_authorizations
+      private
 
-        available_authorizations.select(&:present?)
+      def validate_organization_name_presence
+        translated_attr = "name_#{current_organization.try(:default_locale) || Decidim.default_locale.to_s}".to_sym
+        errors.add(translated_attr, :blank) if send(translated_attr).blank?
       end
 
-      def password
-        encrypted_password.nil? ? super : Decidim::AttributeEncryptor.decrypt(encrypted_password)
-      end
+      def validate_organization_uniqueness
+        base_query = persisted? ? Decidim::Organization.where.not(id:).all : Decidim::Organization.all
 
-      def encrypted_smtp_settings
-        smtp_settings["from"] = set_from
+        organization_names = []
 
-        smtp_settings.merge(encrypted_password: Decidim::AttributeEncryptor.encrypt(password))
-      end
+        base_query.pluck(:name).each do |value|
+          organization_names += value.except("machine_translations").values
+          organization_names += value.fetch("machine_translations", {}).values
+        end
 
-      def set_from
-        return from_email if from_label.blank?
+        organization_names = organization_names.map(&:downcase)
 
-        "#{from_label} <#{from_email}>"
-      end
+        name.each do |language, value|
+          next if value.is_a?(Hash)
 
-      def encrypted_omniauth_settings
-        omniauth_settings.transform_values do |v|
-          Decidim::OmniauthProvider.value_defined?(v) ? Decidim::AttributeEncryptor.encrypt(v) : v
+          errors.add("name_#{language}", :taken) if organization_names.include?(value.downcase)
         end
-      end
 
-      private
-
-      def validate_organization_uniqueness
-        errors.add(:name, :taken) if Decidim::Organization.where(name:).where.not(id:).exists?
+        # name.each do |language, value|
+        #   errors.add(:"name_#{language}", :taken) if Decidim::Organization.where("name::text ilike ?", "%#{value}%").where.not(id:).exists?
+        # end
```

Comment:
O think we can drop these
```suggestion
```

Reply:
Implemented in https://github.com/decidim/decidim/commit/2109c8bfb3cc3736526b211dc9cc0c26636f47ad